src/Core/Router.php

```diff
@@ -39,13 +43,13 @@ private function direct(string $uri, string $methodType)
     private function callAction(string $controller, string $action)
     {
         $controller = "TrendingRepos\\Controller\\{$controller}";
-        $controller = new $controller;
+        $controller = new $controller($this->app);
```

Comment:
To make it work, every controller has to have a constructor with the same signature. Can you enforce this somehow?